File: src/main/java/offer/technical/test/mapper/UserMapper.java
package offer.technical.test.mapper;

import offer.technical.test.model.UserEntity;
import offer.technical.test.model.UserEntityV2;
import offer.technical.test.model.UserResource;
import offer.technical.test.model.UserResourceV2;
import org.mapstruct.Mapper;


@Mapper(componentModel = "spring")
public interface UserMapper {

      UserResource userEntityToUserResource(UserEntity userEntity);

      UserEntity userResourceToUserEntity(UserResource userResource);

      UserResourceV2 userEntityToUserResourceV2(UserEntityV2 userEntity);

      UserEntityV2 userResourceToUserEntityV2(UserResourceV2 userResource);
}

File: src/main/kotlin/offer/technical/test/mapper/UserMapper.kt
package offer.technical.test.mapper

import offer.technical.test.model.UserEntity
import offer.technical.test.model.UserResource
import org.springframework.stereotype.Component

@Component
class UserMapper {

    fun userEntityUserResource(userEntity: UserEntity?): UserResource? {
        if (userEntity == null) {
            return null
        }

        return UserResource(userEntity.name, userEntity.birthDate, userEntity.country, userEntity.phoneNumber, userEntity.gender)
    }

    fun userResourceToUserEntity(userResource: UserResource?): UserEntity? {
        if (userResource == null) {
            return null
        }

        return UserEntity(userResource.name, userResource.birthDate, userResource.country, userResource.phoneNumber, userResource.gender)
    }

}